[WPE] WPE Platform: synthetic clicks generated for unhandled touch events are firing two pointer events

carlosgcampos opened the pull request:
#### 21e718c506bcf8bb3b91621e65c4088fcdcbe4d5
<pre>
[WPE] WPE Platform: synthetic clicks generated for unhandled touch events are firing two pointer events
<a href="https://bugs.webkit.org/show_bug.cgi?id=303531">https://bugs.webkit.org/show_bug.cgi?id=303531</a>

Reviewed by Adrian Perez de Castro.

There are two problems:

  - We should synthesize clicks for taps only when the touch events were not handled.
    So, instead of calling handleGesture right after the touch event is
    passed to the page, we now do it from PageClientImpl::doneWithTouchEvent()
    when wasEventHandled is false.
  - We should notify the web process that synthetic mouse events are a simulated tap.
    When creating the WebMouse Event we check if the input source is
    WPE_INPUT_SOURCE_TOUCHSCREEN to pass WebMouseEventSyntheticClickType::OneFingerTap
    to the constructor.

Canonical link: <a href="https://commits.webkit.org/304226@main">https://commits.webkit.org/304226@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7abdfab036270acb2f055edf7248e8aabd945290

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134947 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7364 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/46195 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142455 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/86800 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7985 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7210 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/103112 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/86800 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137893 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5655 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120954 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83960 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5476 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3084 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/3051 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/114690 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/145155 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7040 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/39686 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/111493 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7093 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/5906 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111845 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5312 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117238 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60959 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20818 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7089 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35409 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/6862 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/70661 "Built successfully") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7096 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6969 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/Shared/wpe/WebEventFactoryWPE.cpp:
```diff
@@ -133,16 +133,21 @@ WebMouseEvent WebEventFactory::createWebMouseEvent(WPEEvent* event)
     auto button = WebMouseEventButton::None;
     auto position = positionFromEvent(event);
     unsigned clickCount = 0;
+    WebMouseEventSyntheticClickType syntheticClickType = WebMouseEventSyntheticClickType::NoTap;
```
Not really important, but in this particular case I find `auto` more readable because it is clear what will be the type of the variable, given that the type of the `enum` is used on the right-hand side of the assignment:

```suggestion
    auto syntheticClickType = WebMouseEventSyntheticClickType::NoTap;
```

aperezdc reviewed (approved):
r=me, neat fix

carlosgcampos committed:
[WPE] WPE Platform: synthetic clicks generated for unhandled touch events are firing two pointer events
https://bugs.webkit.org/show_bug.cgi?id=303531

Reviewed by Adrian Perez de Castro.

There are two problems:

  - We should synthesize clicks for taps only when the touch events were not handled.
    So, instead of calling handleGesture right after the touch event is
    passed to the page, we now do it from PageClientImpl::doneWithTouchEvent()
    when wasEventHandled is false.
  - We should notify the web process that synthetic mouse events are a simulated tap.
    When creating the WebMouse Event we check if the input source is
    WPE_INPUT_SOURCE_TOUCHSCREEN to pass WebMouseEventSyntheticClickType::OneFingerTap
    to the constructor.

Canonical link: https://commits.webkit.org/304226@main
